File: analyze_features.dart
import 'dart:io';

void main() {
  final buyerPath = 'd:/Flutter_Project/food_delivery_app/lib/features/buyer_bloc_architecture';
  final sellerPath = 'd:/Flutter_Project/food_delivery_app/lib/features/seller_bloc_architecture';

  Map<String, List<String>> getFeatureDetails(String basePath) {
    final dir = Directory(basePath);
    if (!dir.existsSync()) return {};
    
    final features = <String, List<String>>{};
    for (final entity in dir.listSync()) {
      if (entity is Directory) {
        final featureName = entity.path.split(Platform.pathSeparator).last;
        final files = <String>[];
        for (final subEntity in entity.listSync(recursive: true)) {
          if (subEntity is File) {
            files.add(subEntity.path.split(Platform.pathSeparator).last);
          }
        }
        features[featureName] = files;
      }
    }
    return features;
  }

  final buyerFeatures = getFeatureDetails(buyerPath);
  final sellerFeatures = getFeatureDetails(sellerPath);

  print("=== Buyer Features ===");
  buyerFeatures.forEach((f, files) {
    print("$f: ${files.length} files");
  });
  
  print("\n=== Seller Features ===");
  sellerFeatures.forEach((f, files) {
    final blocFiles = files.where((x) => x.toLowerCase().contains('bloc') || x.toLowerCase().contains('event') || x.toLowerCase().contains('state')).toList();
    final uiFiles = files.where((x) => x.toLowerCase().contains('ui') || x.toLowerCase().contains('screen') || x.toLowerCase().contains('page')).toList();
    print("$f: ${files.length} files (BLoC: ${blocFiles.length}, UI: ${uiFiles.length})");
  });
}

File: analyze_features.py
import os
import json

def get_feature_details(base_path):
    features = {}
    if not os.path.exists(base_path):
        return features
    for entry in os.listdir(base_path):
        full_path = os.path.join(base_path, entry)
        if os.path.isdir(full_path):
            files = []
            for root, dirs, filenames in os.walk(full_path):
                for f in filenames:
                    files.append(f)
            features[entry] = files
    return features

buyer_path = r"d:\Flutter_Project\food_delivery_app\lib\features\buyer_bloc_architecture"
seller_path = r"d:\Flutter_Project\food_delivery_app\lib\features\seller_bloc_architecture"

buyer_features = get_feature_details(buyer_path)
seller_features = get_feature_details(seller_path)

print("=== Buyer Features ===")
for f, files in buyer_features.items():
    print(f"{f}: {len(files)} files")
    
print("\n=== Seller Features ===")
for f, files in seller_features.items():
    bloc_files = [x for x in files if 'bloc' in x.lower() or 'event' in x.lower() or 'state' in x.lower()]
    ui_files = [x for x in files if 'ui' in x.lower() or 'screen' in x.lower() or 'page' in x.lower()]
    print(f"{f}: {len(files)} files (BLoC: {len(bloc_files)}, UI: {len(ui_files)})")
